mono/metadata/threads.c

```diff
@@ -3587,10 +3587,17 @@ mark_ctx_slots (void *addr, MonoGCMarkFunc mark_func, void *gc_data)
  *   Allocate memory blocks for storing threads or context static data
  */
 static void 
-mono_alloc_static_data (gpointer **static_data_ptr, guint32 offset, gboolean threadlocal)
+mono_alloc_static_data (MonoObject *owner, gpointer **static_data_ptr, guint32 offset, gboolean threadlocal)
```

Comment:
Why not just pass a "size_t owner_id" instead of a "MonoObject *"?
This would eliminate all the branching below.